pkg/engine/journal.go

```diff
@@ -99,6 +100,15 @@ func (entries JournalEntries) Snap(base *deploy.Snapshot) (*deploy.Snapshot, err
 				}
 			case deploy.OpDelete, deploy.OpDeleteReplaced, deploy.OpReadDiscard, deploy.OpDiscardReplaced:
 				if old := e.Step.Old(); !old.PendingReplacement {
+					op := e.Step.Op()
+					contract.Assertf(
+						op == deploy.OpDiscardReplaced || op == deploy.OpReadDiscard ||
+							op == deploy.OpDeleteReplaced || op == deploy.OpDelete,
+						"unexpected step.Op(): %q", op)
```

Comment:
This seems a little overkill, since it's already specified in the `case` statement just two lines above?

Reply:
Yeh copy paste from elsewhere